src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/ProxyService.scala

```diff
@@ -119,6 +119,9 @@ class ProxyService(proxyConfig: ProxyConfig,
           case Some(upgrade) => handleWebSocketRequest(targetHost, request, upgrade)
           case None => handleHttpRequest(targetHost, request)
         }
+        responseFuture map {response =>
```

Comment:
I don't think the futures are hooked up correctly here. Should be something like:
```
        responseFuture map { response =>
          logRequestForMetrics(userInfo, request, response)
          response
        } recover { case e =>
          logger.error("Error occurred in Jupyter proxy", e)
          throw ProxyException(googleProject, clusterName)
        }
```

Reply:
right, thanks